src/CommunityToolkit.Maui.MediaElement/Views/MediaManager.android.cs

```diff
@@ -11,12 +11,16 @@
 using CommunityToolkit.Maui.Views;
 using Microsoft.Extensions.Logging;
 
+
 namespace CommunityToolkit.Maui.Core.Views;
 
 public partial class MediaManager : Java.Lang.Object, IPlayer.IListener
 {
 	double previousSpeed = -1;
 	float volumeBeforeMute = 1;
+	readonly Page currentPage = Application.Current?.MainPage ?? throw new NullReferenceException();
```

Comment:
As I mentioned on twitter you need to use the PageExtension method. Because the `MainPage` can be a `NavigationPage` and I believe that you want the `ContentPage` that is presented instead.

Reply:
I am going to rewrite it as you suggested. Sorry I did not get back to you about that earlier. I have been working on removing code duplication.

Reply:
Let me know if need anything (:

Reply:
With Help from Pedro It should be fixed now. It has been updated as you suggested to use Page Extensions and is working for windows and android now.